testing/calibration/contract/fil-gas-calibration-actor/src/lib.rs

```diff
@@ -244,14 +294,22 @@ fn random_mutations(data: &mut Vec<u8>, seed: u64, n: usize) {
     }
 }
 
+/// Generates a random string in the 0x20 - 0x7e ASCII character range
+/// (alphanumeric + symbols, excluding the delete symbol).
+fn random_ascii_string(n: usize, seed: u64) -> String {
+    let bytes = lcg64(seed).map(|x| ((x % 95) + 32) as u8).take(n).collect();
+    String::from_utf8(bytes).unwrap()
+}
+
 /// Knuth's quick and dirty random number generator.
 /// https://en.wikipedia.org/wiki/Linear_congruential_generator
-fn lcg64(mut seed: u64) -> impl Iterator<Item = u64> {
+fn lcg64(initial_seed: u64) -> impl Iterator<Item = u64> {
     let a = 6364136223846793005;
     let c = 1442695040888963407;
+    let mut seed = Wrapping(initial_seed);
     std::iter::repeat_with(move || {
-        seed = a * seed + c;
-        seed
+        seed = Wrapping(a) * seed + Wrapping(c);
```

Comment:
I'm not entirely sure why, but the lack of wrapping arithmetics here caused weird side effects during ASCII string generation, which led to weird characters. This is more correct.